golem-worker-executor-base/src/worker.rs

```diff
@@ -422,7 +427,7 @@ impl<Ctx: WorkerCtx> Worker<Ctx> {
         calling_convention: CallingConvention,
         full_function_name: String,
         function_input: Vec<Value>,
-    ) -> Result<Option<Result<Vec<Value>, GolemError>>, GolemError> {
+    ) -> Result<Option<Result<TypeAnnotatedValue, GolemError>>, GolemError> {
```

Comment:
As discussed in private, I think this should be `Vec<TypeAnnotatedValue>` everywhere  where we had `Vec<Value>`

Reply:
For others: We both confirmed why it is TypeAnnotatedValue